src/lib.rs

```diff
@@ -503,64 +503,83 @@ impl Arbitrary for AtomicUsize {
     }
 }
 
-impl<A> Arbitrary for Range<A>
-where
-    A: Arbitrary + Clone + PartialOrd,
-{
-    fn arbitrary(u: &mut Unstructured<'_>) -> Result<Self> {
-        let (start, end): (A, A) = Arbitrary::arbitrary(u)?;
-        Ok(to_range(start, end, |a, b| a..b))
-    }
+macro_rules! impl_range {
+    ($range:ty, $value_closure:expr, $value_ty:ty, $fun:ident($fun_closure:expr)) => {
+        impl<A> Arbitrary for $range
+        where
+            A: Arbitrary + Clone + PartialOrd,
+        {
+            fn arbitrary(u: &mut Unstructured<'_>) -> Result<Self> {
+                let value: $value_ty = Arbitrary::arbitrary(u)?;
+                Ok($fun(value, $fun_closure))
+            }
 
-    #[inline]
-    fn size_hint(depth: usize) -> (usize, Option<usize>) {
-        crate::size_hint::and(
-            <A as Arbitrary>::size_hint(depth),
-            <A as Arbitrary>::size_hint(depth),
-        )
-    }
+            #[inline]
+            fn size_hint(depth: usize) -> (usize, Option<usize>) {
+                crate::size_hint::and(
+                    <A as Arbitrary>::size_hint(depth),
+                    <A as Arbitrary>::size_hint(depth),
+                )
```

Comment:
This should be just `<A as Arbitrary>::size_hint(depth)` for unbounded ranges, so I think this macro needs to be tweaked a little bit.

Reply:
Ooppsss...